pkg/k8shandler/reconciler.go

```diff
@@ -54,5 +59,112 @@ func Reconcile(requestCluster *elasticsearch.Elasticsearch, requestClient client
 		return fmt.Errorf("Failed to reconcile PrometheusRules for Elasticsearch cluster: %v", err)
 	}
 
+	// Ensure index management is in place
+	if err := elasticsearchRequest.CreateOrUpdateIndexManagement(); err != nil {
+		return fmt.Errorf("Failed to reconcile IndexMangement for Elasticsearch cluster: %v", err)
+	}
+
+	return nil
+}
+
+func (req *ElasticsearchRequest) CreateIndex(name string, index *estypes.Index) error {
```

Comment:
I feel like these new functions should go into `k8shandler/elasticsearch.go`

Reply:
moved